### PR TITLE
store/tikv: add safeTime config for gc worker.

### PR DESCRIPTION
**Pull request by disksing**
@shenli @coocood @zimulala @tiancaiamao

### CODE REVIEW

File: store/tikv/gc_worker.go [hunk 1]
```diff
@@ -80,33 +101,115 @@ func (w *GCWorker) start(ver uint64) {
...
 				break
 			}
-			if isLeader && !gcIsRunning {
-				if w.store.oracle.IsExpired(safePoint, uint64(gcRunInterval/time.Millisecond)) {
-					gcIsRunning = true
-					go w.runGCJob(safePoint)
+			if isLeader {
+				err = w.leaderTick()
+				if err != nil {
+					log.Warnf("[gc worker] leader tikc err: %v", err)
```
Comment (shenli) [inline store/tikv/gc_worker.go]: tikc -> tick

File: store/tikv/gc_worker.go [hunk 2]
```diff
@@ -80,33 +101,115 @@ func (w *GCWorker) start(ver uint64) {
...
+		return errors.Trace(err)
+	}
+
+	w.gcIsRunning = true
 	log.Infof("[gc worker] %s starts GC job, safePoint: %v", w.uuid, safePoint)
+	go w.runGCJob(safePoint)
+	return nil
+}
+
+func (w *GCWorker) prepareGC() (bool, uint64, error) {
```
Comment (shenli) [inline store/tikv/gc_worker.go]: Add comment for this function.

Comment (shenli) [inline store/tikv/gc_worker.go]: prepare() is enough.

File: store/tikv/gc_worker.go [hunk 3]
```diff
@@ -80,33 +101,115 @@ func (w *GCWorker) start(ver uint64) {
...
+	now, err := w.getOracleTime()
+	if err != nil {
+		return false, 0, errors.Trace(err)
+	}
+	ok, err := w.checkGCInterval(now)
+	if err != nil || !ok {
+		return false, 0, errors.Trace(err)
+	}
+	safePoint, err := w.calculateSafePoint(now)
```
Comment (coocood) [inline store/tikv/gc_worker.go]: Name it `newSafePoint` and `calculateNewSafePoint` is easer to read.

**Comment by disksing** [comment]
PTAL @shenli @coocood

File: store/tikv/gc_worker.go [hunk 4]
```diff
@@ -80,33 +101,117 @@ func (w *GCWorker) start(ver uint64) {
...
+	}
+
+	w.gcIsRunning = true
 	log.Infof("[gc worker] %s starts GC job, safePoint: %v", w.uuid, safePoint)
+	go w.runGCJob(safePoint)
+	return nil
+}
+
+// prepare checks required conditions for starting a GC job. It returns a bool
+// that indicates whether the GC job should start and the new safePoint.
```
Comment (tiancaiamao) [inline store/tikv/gc_worker.go]: s/and/at ??

Comment (tiancaiamao) [inline store/tikv/gc_worker.go]: ignore me...

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**